src/parser.c

```diff
@@ -596,6 +596,21 @@ void read_parameter_list_decl(func_t *func, int anon)
 {
     int vn = 0;
     lex_expect(T_open_bracket);
+
+    char token[MAX_TYPE_LEN];
+    if (lex_peek(T_identifier, token) && !strncmp(token, "void", 4)) {
+        next_token = lex_token();
+        if (lex_accept(T_close_bracket))
+            return;
+        strcpy(func->param_defs[vn].type_name, token);
```

Comment:
Latest change (#201) deprecated `var_t#type_name`. Instead, you should use `read_full_var_decl` to properly initialize the parameter `var_t` then check if `func->param_defs[vn].type == TY_void` to assert later criteria.

Reply:
`read_full_var_decl` cannot handle both cases of named and unnamed.  If `anon` is zero, it expects an identifier and abort on void parameter lists; if `anon` is non-zero, it doesn't correctly read the first parameter with types such as `void *`.  The original implementation is actually a custom version of `read_partial_var_decl`.

Calling the function twice with different `anon` set can handle legal cases, but will allow things like `void [] name []`.  So, the solution seems same but with a newer way to construct `var_t`.